src/main/scala/ai/privado/audit/DataElementDiscovery.scala

```diff
@@ -0,0 +1,193 @@
+
+  // Get list of Class Name having getter and setter method
+  def getSourceUsingRules(xtocpg: Try[Cpg]): List[String] = {
+    logger.info("Process Class Name from cpg")
+    val classNameList = ListBuffer[String]()
+    xtocpg match {
+      case Success(cpg) => {
+        // Get DTO/Entity Class name
+        val typeDeclList = cpg.typeDecl
+          .filter(_.order > 0)
+          .whereNot(_.name("^(.*)(Controller|Service|Impl|Helper|Util|Processor)$"))
+          .or(
+            _.where(_.method.name("^(get|set).*")),
+            _.where(_.method.name("^(hascode|equals)")),
+            _.where(_.annotation.name(".*(Getter|Setter).*"))
+          )
+          .toList
+        typeDeclList.foreach(node => {
+          if (node.head.fullName.nonEmpty) {
+            classNameList += node.fullName
+          }
+        })
+      }
+      case Failure(exception) => {
+        println("Failed to process class name from cpg")
+        logger.debug("Failed to process class name from cpg", exception)
+        println(exception.printStackTrace())
+      }
+    }
+    logger.info("Successfully Processed Class Name from cpg")
+    classNameList.toList
+  }
+
+  // Search other potential Class in package
+  def extractClassFromPackage(xtocpg: Try[Cpg], classFullNameList: Set[String]): Set[String] = {
+    val packageNameSet = mutable.Set[String]()
+
+    classFullNameList.foreach(fullName => {
+      val lastDotIndex = fullName.lastIndexOf('.')
+      if (lastDotIndex > 0) {
+        val packageName = fullName.substring(0, lastDotIndex)
+        packageNameSet += packageName
+      }
+    })
+
+    val derivedClassName = ListBuffer[String]()
+    xtocpg match {
+      case Success(cpg) => {
+        packageNameSet.foreach(packageName => {
+          val pattern = s"$packageName.*"
+          val typeDeclList = cpg.typeDecl
+            .filter(_.order > 0)
+            .where(_.fullName(pattern))
+            .whereNot(_.name("^(.*)(Controller|Service|Impl|Helper|Util|Processor)$"))
+            .toList
+          typeDeclList.foreach(typeDecl => derivedClassName += typeDecl.fullName)
+        })
+      }
+      case Failure(exception) => {
+        println("Failed to Extract package classes from cpg")
+        logger.debug("Failed to Extract package classes from cpg", exception)
+        println(exception.printStackTrace())
+      }
+    }
+    derivedClassName.toSet
+  }
+
+  // Get list of member variable present in given class
+  def getMemberUsingClassName(xtocpg: Try[Cpg], classNameSet: Set[String]): mutable.Map[String, List[Member]] = {
+    logger.info("Process Member Name from cpg")
+    val memberInfoMap = mutable.Map[String, List[Member]]()
+
+    xtocpg match {
+      case Success(cpg) => {
+        classNameSet.foreach(className => {
+          // Get member variable of class and put it into map
+          // val memberInfo = cpg.member.filter(_.typeDecl.order > 0).where(_.typeDecl.fullName(className)).l
```

Comment:
Get rid of this commented code if not required.

Reply:
Yes Done.